lib/sprite_animation.dart

```diff
@@ -102,21 +105,21 @@ class SpriteAnimation {
     int amount,
     List<double> stepTimes, {
     int amountPerRow,
-    double textureX = 0.0,
-    double textureY = 0.0,
-    double textureWidth,
-    double textureHeight,
+    Vector2 texturePosition,
+    Vector2 textureSize,
     this.loop = true,
   }) : assert(amountPerRow == null || amount >= amountPerRow) {
     amountPerRow ??= amount;
     frames = List<SpriteAnimationFrame>(amount);
     for (var i = 0; i < amount; i++) {
+      final position = Vector2(
+        texturePosition.x + (i % amountPerRow) * textureSize.x,
+        texturePosition.y + (i ~/ amountPerRow) * textureSize.y,
```

Comment:
ditto. also, can this be a method? it's repeated twice

Reply:
The DRY problem is solved on my PR about asset loading

Reply:
I converted them, but swapped back. I think it is easier to read like it is now compared to this:
```dart
      final position = texturePosition +
          Vector2(i % amountPerRow.toDouble(), (i ~/ amountPerRow).toDouble())
            ..multiply(textureSize);
```

Reply:
true, I forgot the toDouble. we should make a Vector2.fromInts I think (like we had for Position). but that is a followup for sure.